tests/test_cli/test_launch.py

```diff
@@ -16,32 +16,76 @@
 #   limitations under the License.
 #
 # ------------------------------------------------------------------------------
-
 """This test module contains the tests for the `aea launch` sub-command."""
 
 import logging
 import os
 import shutil
-import signal
-import subprocess  # nosec
 import sys
 import tempfile
 import time
+import unittest
+from contextlib import contextmanager
 from pathlib import Path
+from typing import Generator, List, Optional
+
+from pexpect.exceptions import EOF  # type: ignore
 
 import yaml
 
 from aea.cli import cli
 from aea.configurations.base import DEFAULT_AEA_CONFIG_FILE
 from aea.test_tools.click_testing import CliRunner
 
+from tests.common.pexpect_popen import PexpectSpawn
+
 from ..conftest import AUTHOR, CLI_LOG_OPTION, CUR_PATH
 
 logger = logging.getLogger(__name__)
 
 
-class TestLaunch:
-    """Test that the command 'aea launch <agent_name>' works as expected."""
+class BaseLaunchTestCase:
+    """Base Test case for launch tests."""
+
+    @classmethod
+    def setUpClass(cls) -> None:
```

Comment:
I think we should stick to `snack_case`

Reply:
it's python unit test naming:
https://docs.python.org/3/library/unittest.html
but will switch to pytest's style.

Reply:
hmm ok